### PR TITLE
NAS-103711 ix-auto property

### PR DESCRIPTION
**Pull request by dkmullen**
Adds (via a directive) a unique identifier, ix-auto, to as many elements in the UI as possible. Also fixes some broken ids in the tables. This branch should work behind the scenes without making any visible changes in the UI

### CODE REVIEW

**Comment by erinix** [comment]
could you please test this @wblock ?

**Comment by wblock** [comment]
This looks very good to me.  The only thing I saw that was a little long was "clickable_overlay".  I'd prefer just "overlay" to make the labels shorter, although it's not a big problem.  This is something we can refine, assuming it doesn't require editing in a lot of places to change.
At present, I don't use the ix-auto-type property, but other automation might find them useful.

**Comment by wblock** [comment]
@ericbsd What do you think--will this cause problems for you?

**Comment by ericbsd** [comment]
> @ericbsd What do you think--will this cause problems for you?

As far as I see, I don't think it will cause much issue.

File: src/app/pages/common/entity/entity-table/entity-table.component.html [hunk 1]
```diff
@@ -245,6 +268,10 @@
             <div class="clickable"
               *ngIf="col.toggle"
               id="{{row.name}}_{{col.name}}-overlay"
+              [ix-auto]="col?.name"
+              ix-auto-type="clickable-overlay"
```
Comment (wblock) [inline src/app/pages/common/entity/entity-table/entity-table.component.html]: Shortening this seems fine to me:
```suggestion
              ix-auto-type="overlay"
```

Comment (dkmullen) [reply src/app/pages/common/entity/entity-table/entity-table.component.html]: done!

**Pull request merged**